packages/notifications/src/elements/toaster/Toast.tsx

```diff
@@ -0,0 +1,71 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+import { IToast } from './reducer';
+import { useToast } from './useToast';
+
+interface IToastProps {
+  toast: IToast;
+  pauseTimers: boolean;
+}
+
+export const Toast: React.FC<IToastProps> = ({ toast, pauseTimers }) => {
+  const { removeToast } = useToast();
+  const { id } = toast;
+  const { autoDismiss } = toast.options;
+
+  const [remainingTime, setRemainingTime] = useState(NaN);
+  const startTimeRef = useRef(Date.now());
+  const previousRemainingTimeRef = useRef(remainingTime);
+
+  /**
+   * Reset timer if `autoDimiss` option is updated
+   */
+  useEffect(() => {
+    if (typeof autoDismiss === 'number') {
+      setRemainingTime(autoDismiss);
+    } else {
+      setRemainingTime(NaN);
+    }
+  }, [autoDismiss]);
+
+  /**
+   * Store remaining time when `pauseTimers` is provided
+   */
+  useEffect(() => {
+    if (pauseTimers && !isNaN(remainingTime)) {
+      previousRemainingTimeRef.current = remainingTime - (Date.now() - startTimeRef.current);
+
+      setRemainingTime(NaN);
+    } else if (!pauseTimers && isNaN(remainingTime) && !isNaN(previousRemainingTimeRef.current)) {
+      setRemainingTime(previousRemainingTimeRef.current);
+    }
+  }, [pauseTimers, remainingTime]);
+
+  useEffect(() => {
+    let timeout: any;
+
+    if (!isNaN(remainingTime)) {
+      startTimeRef.current = Date.now();
+
+      timeout = setTimeout(() => {
```

Comment:
I learned recently that using `window.setTimeout` allows TypeScript to infer the timeout id (`number`). I think it could enable the removal of the `any` type on line 51.

```tsx
let timeout: number | undefined = undefined;

timeout = window.setTimeout(callback, delay);
```

Reply:
Aren't there SSR concerns for dereferencing `window`?

Reply:
Yeah, that's a good point. SSR support slipped my mind. ⛔ `window` reference.